src/Lecture1.hs

```diff
@@ -50,7 +50,11 @@ Explanation: @sumOfSquares 3 4@ should be equal to @9 + 16@ and this
 is 25.
 -}
 -- DON'T FORGET TO SPECIFY THE TYPE IN HERE
-sumOfSquares x y = error "TODO!"
+
+-- I got the following warning not sure not to properly fix it
+-- Defaulting the following constraints to type ‘Integer’      (Integral b0)
+sumOfSquares :: Num a => a -> a -> a
+sumOfSquares x y = x^(2 :: Integer) + y^(2 :: Integer)
```

Comment:
This is the correct solution 👍🏻 And that's one way to remove this warning.

The Haskell compiler outputs some warnings because the operator `^` is polymorphic on both arguments and numeric literals are polymorphic as well. 

This is unfortunately 😞 But it's a problem when using the `^` operator. One of the possible solutions is to specify the type of constants 2 to some specific type using type annotations:

```suggestion
sumOfSquares x y = x ^ (2 :: Int) + y ^ (2 :: Int)
```

I agree that it looks a bit ugly now 😕 

Another approach is to use `*` in this particular case to avoid ambiguity:

```suggestion
sumOfSquares x y = x * x + y * y
```